File: iOS/UITableView/UITableView/Expense.swift
import UIKit

struct Expense {
    let date: String
    let amount: String
    let category: String
    let categoryIcon: UIImage?
    let paymentMethod: UIImage?
}

extension Expense {
    static func createDummy() -> [Expense] {
        return [Expense(date: "2022-11-01 19:30", amount: "￥1,000", category: "Groceries", categoryIcon: UIImage(named: "groceries"), paymentMethod: UIImage(named: "cash")),
                Expense(date: "2022-11-02 15:30", amount: "￥5,000", category: "Electricity", categoryIcon: UIImage(named: "electricity"), paymentMethod: UIImage(named: "cash")),
                Expense(date: "2022-11-03 15:33", amount: "￥15,000", category: "Clothing", categoryIcon: UIImage(named: "clothing"), paymentMethod: UIImage(named: "credit-card")),
                Expense(date: "2022-11-04 20:33", amount: "￥50,000", category: "Medical", categoryIcon: UIImage(named: "medical"), paymentMethod: UIImage(named: "credit-card")),
                Expense(date: "2022-11-05 15:25", amount: "￥90,000", category: "Eating Out", categoryIcon: UIImage(named: "eating-out"), paymentMethod: UIImage(named: "credit-card")),
                Expense(date: "2022-11-06 16:00", amount: "￥55,000", category: "Beauty", categoryIcon: UIImage(named: "beauty"), paymentMethod: UIImage(named: "cash")),
                Expense(date: "2022-11-07 11:00", amount: "￥995,000", category: "Other", categoryIcon: UIImage(named: "other"), paymentMethod: UIImage(named: "credit-card")),
                Expense(date: "2022-11-08 09:00", amount: "￥5,000", category: "Household Supplies", categoryIcon: UIImage(named: "household-supplies"), paymentMethod: UIImage(named: "cash")),
                Expense(date: "2022-11-09 09:10", amount: "￥5,000", category: "Transportation", categoryIcon: UIImage(named: "transportation"), paymentMethod: UIImage(named: "cash")),
                Expense(date: "2022-11-10 22:50", amount: "￥2,000", category: "Water", categoryIcon: UIImage(named: "water"), paymentMethod: UIImage(named: "cash")),
                Expense(date: "2022-11-11 12:40", amount: "￥3,000", category: "Recreation", categoryIcon: UIImage(named: "recreation"), paymentMethod: UIImage(named: "cash")),
                Expense(date: "2022-11-12 11:40", amount: "￥4,000", category: "Groceries", categoryIcon: UIImage(named: "groceries"), paymentMethod: UIImage(named: "credit-card")),
                Expense(date: "2022-11-13 10:45", amount: "￥5,400", category: "Groceries", categoryIcon: UIImage(named: "groceries"), paymentMethod: UIImage(named: "credit-card")),
                Expense(date: "2022-11-14 16:50", amount: "￥6,000", category: "Eating Out", categoryIcon: UIImage(named: "eating-out"), paymentMethod: UIImage(named: "credit-card")),
                Expense(date: "2022-11-15 16:55", amount: "￥7,000", category: "Gas", categoryIcon: UIImage(named: "gas"), paymentMethod: UIImage(named: "cash")),
                Expense(date: "2022-11-16 11:55", amount: "￥8,000", category: "Groceries", categoryIcon: UIImage(named: "groceries"), paymentMethod: UIImage(named: "cash")),
                Expense(date: "2022-11-17 08:55", amount: "￥11,000", category: "Gas", categoryIcon: UIImage(named: "gas"), paymentMethod: UIImage(named: "cash")),
                Expense(date: "2022-11-18 18:25", amount: "￥4,000", category: "Clothing", categoryIcon: UIImage(named: "clothing"), paymentMethod: UIImage(named: "cash")),
                Expense(date: "2022-11-19 12:55", amount: "￥5,000", category: "Groceries", categoryIcon: UIImage(named: "groceries"), paymentMethod: UIImage(named: "credit-card")),
                Expense(date: "2022-11-20 12:48", amount: "￥6,000", category: "Household Supplies", categoryIcon: UIImage(named: "household-supplies"), paymentMethod: UIImage(named: "cash")),
                Expense(date: "2022-11-21 11:18", amount: "￥7,000", category: "Household Supplies", categoryIcon: UIImage(named: "household-supplies"), paymentMethod: UIImage(named: "cash")),
                Expense(date: "2022-11-22 10:00", amount: "￥5,700", category: "Eating Out", categoryIcon: UIImage(named: "eating-out"), paymentMethod: UIImage(named: "cash")),
                Expense(date: "2022-11-23 12:00", amount: "￥5,700", category: "Groceries", categoryIcon: UIImage(named: "groceries"), paymentMethod: UIImage(named: "cash")),
                Expense(date: "2022-11-24 19:11", amount: "￥1,300", category: "Groceries", categoryIcon: UIImage(named: "groceries"), paymentMethod: UIImage(named: "credit-card")),
                Expense(date: "2022-11-25 11:11", amount: "￥534,000", category: "Groceries", categoryIcon: UIImage(named: "groceries"), paymentMethod: UIImage(named: "credit-card")),
                Expense(date: "2022-11-26 16:19", amount: "￥524,000", category: "Eating Out", categoryIcon: UIImage(named: "eating-out"), paymentMethod: UIImage(named: "credit-card")),
                Expense(date: "2022-11-27 16:19", amount: "￥54,500", category: "Groceries", categoryIcon: UIImage(named: "groceries"), paymentMethod: UIImage(named: "credit-card")),
                Expense(date: "2022-11-28 10:29", amount: "￥5,500", category: "Groceries", categoryIcon: UIImage(named: "groceries"), paymentMethod: UIImage(named: "credit-card")),
                Expense(date: "2022-11-29 11:45", amount: "￥4,000", category: "Eating Out", categoryIcon: UIImage(named: "eating-out"), paymentMethod: UIImage(named: "credit-card")),
                Expense(date: "2022-11-30 15:45", amount: "￥1,000", category: "Groceries", categoryIcon: UIImage(named: "groceries"), paymentMethod: UIImage(named: "cash"))]
    }
}

File: Android/RecyclerView/app/src/main/java/tokyo/randx/portfolio/android/recyclerview/Expense.java
package tokyo.randx.portfolio.android.recyclerview;

import android.widget.ImageView;

import java.util.ArrayList;
import java.util.List;

public class Expense {
    String category;
    String date;
    String memo;
    String paymentMethod;
    Integer amount;

    Expense(String category, String date, String memo, String paymentMethod, Integer amount) {
        this.category = category;
        this.date = date;
        this.memo = memo;
        this.paymentMethod = paymentMethod;
        this.amount = amount;
    }
    public static List<Expense> createDummy() {
        List<Expense> expenses = new ArrayList<>();
        Expense e1 = new Expense("groceries", "2023-01-01 08:00", "イーオン", "visa", 2500);
        Expense e2 = new Expense("automotive", "2023-01-02 08:30", "車検", "jcb", 15000);
        Expense e3 = new Expense("beauty", "2023-01-03 09:30", "美容院", "cash", 8000);
        Expense e4 = new Expense("cellphone", "2023-01-04 11:30", "携帯料金", "visa", 3000);
        Expense e5 = new Expense("clothing", "2023-01-05 15:00", "UNIQLO", "visa", 5600);
        Expense e6 = new Expense("eating_out", "2023-01-06 14:30", "外食", "visa", 4356);
        Expense e7 = new Expense("electricity", "2023-01-07 10:35", "電気代", "cash", 5690);
        Expense e8 = new Expense("gas", "2023-01-08 16:30", "ガス代", "cash", 3680);
        Expense e9 = new Expense("internet", "2023-01-09 17:30", "光回線", "visa", 4800);
        Expense e10 = new Expense("medical", "2023-01-10 19:30", "歯医者", "visa", 1200);
        Expense e11 = new Expense("transportation", "2023-01-11 09:36", "定期代", "visa", 22780);
        Expense e12 = new Expense("water", "2023-01-12 20:15", "水道料金", "cash", 6800);

        expenses.add(e1);
        expenses.add(e2);
        expenses.add(e3);
        expenses.add(e4);
        expenses.add(e5);
        expenses.add(e6);
        expenses.add(e7);
        expenses.add(e8);
        expenses.add(e9);
        expenses.add(e10);
        expenses.add(e11);
        expenses.add(e12);

        return expenses;
    }
}
